### PR TITLE
E2E: streamline specs for legacy WordPress.com plans.

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

This PR is a housekeeping change to streamline existing specs for the WordPress.com legacy plans.

Key changes:
- removal of redundant spec `Plans (Legacy): Upgrade`;
- refactor the `SnackbarNoticeComponent` into `NoticeComponent` in order to make it more generic;
- added step to Legacy Plan spec to validate the presence of a notice banner;

#### Testing instructions

Ensure the following:
  - [x] Pre-Release E2E

Related to https://github.com/Automattic/wp-calypso/issues/62492.
Depends on https://github.com/Automattic/wp-calypso/pull/63974.

### CODE REVIEW

**Comment by worldomonation** [comment]
Ready to review, but putting it on hold as a Draft until the parent in the stack is reviewed and merged.

File: packages/calypso-e2e/src/lib/components/notice-component.ts
Comment (dpasque) [inline packages/calypso-e2e/src/lib/components/notice-component.ts]: 100% optional! The original implementation was this way, so this may be out of scope for this PR. But wanted to throw out a suggestion for this function while we're here! 🙂 

The way I see it, we've traditionally written two kinds of functions:
1. True boolean functions that use `count` to see if something is present in the DOM, and returning a boolean value based on that.
2. `Validate` or `WaitFor` functions that use `waitFor` on an element, return void, and throw an error if it's not there.

Option 2 is used more when we know there may be an async gap between something happening and the expected element showing up.

I think we may want to make this function fit into one of those two patterns. The reason is that this function technically can never return false! If the element isn't there, `waitFor` will throw. So it will either throw or return true. I think that combo might confuse naive callers, who expect a throw-safe boolean value.

**Review by dpasque — APPROVED** [review]
Not sure how much you wanted to add before switching from draft, but I really like this cleanup! 👍 

Approving for now - let me know if there's any other big changes later you want me to look at. 🙂 

Also, left 1 optional comment that's probably out of scope for this PR, but threw it out there anyway in case you thought now might be a good time for the update.

**Comment by worldomonation** [comment]
> Not sure how much you wanted to add before switching from draft, but I really like this cleanup! 👍

I'll rebase and re-run the specs to make sure things are OK before marking it as ready. 

In the meantime, thanks for pointing out the method won't ever return values other than True - I totally missed that and will think of a solution.

File: packages/calypso-e2e/src/lib/components/notice-component.ts
Comment (worldomonation) [reply packages/calypso-e2e/src/lib/components/notice-component.ts]: You're 100% right - I am not sure why I was returning a boolean value from this method when the calling test steps don't even use it to assert. Then there's the obvious point about this method never actually returning a false value.

I blame my lack of focus when I pushed up this change!

Comment (worldomonation) [reply packages/calypso-e2e/src/lib/components/notice-component.ts]: Revised in ﻿﻿10977cded6feeaa28befa9025623a11c6fed2ea4: 
- no longer return a Boolean value but instead throw.

**Pull request merged**
